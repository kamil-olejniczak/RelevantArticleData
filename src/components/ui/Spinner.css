.Spinner{
    text-align: center;
    color: #fff;
}

.beat-container{
    position: relative;
    width: 5%;
    margin: 0 auto;
    display: table;
    height: 20px;
}

.beat-loader{
    display: table-cell;
}

.beat-top{
    top: 0;
}

.beat-bottom{
    bottom: 0;
}

.beat{
    height: 10px;
    width: 5px;
    display: block;
    position: absolute;
    background-color: #93228D;
    animation: beat_loader  1.5s  infinite ease-in-out;
}

.beat:nth-child(1){
    left: 10px;
    animation-delay: .2s;
}

.beat:nth-child(2){
    left: 16px;
    animation-delay: .3s;
}
.beat:nth-child(3){
    left: 22px;
    animation-delay: .4s;
}
.beat:nth-child(4){
    left: 28px;
    animation-delay: .5s;
}
.beat:nth-child(5){
    left: 34px;
    animation-delay: .6s;
}
.beat:nth-child(6){
    left: 40px;
    animation-delay: .7s;
}

@keyframes beat_loader {
    0% {
        transform: translateY(0px);
        height: 10px;
        background-color: #F20075;
    }
    25% {
        transform: translateY(-20px);
        height: 50px;
        background-color: #B84B9E;
    }
    50% {
        transform: translateY(0px);
        height: 10px;
        background-color: #FF6EB0;
    }
    100% {
        transform: translateY(0px);
        height: 10px;
        background-color: #F20075;
    }
}