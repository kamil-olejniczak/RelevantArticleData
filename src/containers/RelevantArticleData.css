.dataDiv {
    float: right;
    margin: 20px auto 0;
    padding: 10px 0;
    width: 25%;
    background-color: white;
    border-radius: 5px;
}

@media only screen and (max-width: 1670px) {
    .dataDiv {
        float: none;
        display: block;
        margin-bottom: 15px;
        width: 70%;
        max-width: 400px;
    }
}