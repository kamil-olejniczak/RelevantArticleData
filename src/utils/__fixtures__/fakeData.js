export const firstRecentlyVisitedSite = {
    id: 1,
    title: 'SomeTitle',
    url: 'SomeUrl',
    date: new Date(1 << 30),
};
export const secondRecentlyVisitedSite = {
    id: 2,
    title: 'SomeTitle2',
    url: 'SomeUrl2',
    date: new Date(1 << 30),
};
export const fakeRecentlyVisited = {
    recentlyVisited: [firstRecentlyVisitedSite, secondRecentlyVisitedSite]
};
export const fakeArticleData = {
    data: {
        title: 'Some Title',
        date_published: '1970-01-01',
        domain: 'www.dabliu.com',
        dek: 'Short dek text',
        excerpt: 'Short excerpt text',
        word_count: 111,
        lead_image_url: 'http://www.dabliu.pl/image.html',
    }
};
export const fakeTagStatistics = {
    div: 8,
    p: 2,
    a: 3,
    span: 10,
    i: 9,
    html: 1,
    br: 2,
    table: 7,
    hr: 6,
    input: 4,
    textarea: 5
};
export const fakeSortedTagStatistics = [
    'span', '10',
    'i', '9',
    'div', '8',
    'table', '7',
    'hr', '6',
    'textarea', '5',
    'input', '4',
    'a', '3',
    'br', '2',
    'p', '2',
];