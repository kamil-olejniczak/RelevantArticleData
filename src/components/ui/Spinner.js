import React from 'react';
import './Spinner.css';

const Spinner = () => (
    <div className="Spinner">
        <div className="beat-container">
            <div className="beat-loader">
                <div className="beat beat-top"></div>
                <div className="beat beat-top"></div>
                <div className="beat beat-top"></div>
                <div className="beat beat-top"></div>
                <div className="beat beat-top"></div>
                <div className="beat beat-top"></div>
            </div>
        </div>
    </div>
);

export default Spinner;