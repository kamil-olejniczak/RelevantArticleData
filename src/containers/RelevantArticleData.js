import React, {Component} from 'react';
import TagStatistics from '../components/article/TagStatistics';
import RecentlyVisitedArticles from '../components/article/RecentlyVisitedArticles';
import {connect} from 'react-redux';
import ParsedArticle from '../components/article/ParsedArticle';
import ArticleSearch from './ArticleSearch';
import './RelevantArticleData.css';
import ArticleData from '../components/article/ArticleData';
import * as articleActions from '../store/actions/article';
import * as localStorageActions from '../store/actions/localStorage';
import withNavbar from '../components/ui/navigation/Navbar';
import Spinner from "../components/ui/Spinner";

class RelevantArticleData extends Component {
    render() {
        return (
            <div>
                <ArticleSearch/>
                {this.props.isArticleShown ? (
                    <div>
                        <ArticleData data={this.props.data} isModalClosedByUser={this.props.isModalClosedByUser}/>
                        <ParsedArticle content={this.props.content}/>
                        <div className="dataDiv">
                            <RecentlyVisitedArticles recentlyVisited={this.props.recentlyVisited}
                                                     clearData={this.props.clearData}/>
                            <TagStatistics tagStats={this.props.statistics}/>
                        </div>
                    </div>) : null}
                {this.props.isDataBeingResolved ? (<Spinner/>) : null}
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        data: state.article.data,
        content: state.article.content,
        recentlyVisited: state.article.recentlyVisited,
        statistics: state.article.statistics,
        isArticleShown: state.article.isArticleShown,
        isModalClosedByUser: state.article.isModalClosedByUser,
        isDataBeingResolved: state.article.isDataBeingResolved
    };
};

const mapDispatchToProps = dispatch => {
    return {
        clearArticle: () => dispatch(articleActions.clearArticle()),
        clearData: () => dispatch(localStorageActions.clearData())
    };
};

export const PureRelevantArticleData = (withNavbar(RelevantArticleData));
export default connect(mapStateToProps, mapDispatchToProps)(withNavbar(RelevantArticleData));