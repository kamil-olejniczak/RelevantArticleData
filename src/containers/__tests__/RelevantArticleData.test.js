import * as React from 'react';
import configureStore from 'redux-mock-store';
import {MemoryRouter} from 'react-router-dom';
import RelevantArticleData, {PureRelevantArticleData} from '../RelevantArticleData';
import {render, shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Spinner from '../../components/ui/Spinner';
import {fakeArticleData, fakeRecentlyVisited} from '../../utils/__fixtures__/fakeData';

describe('DetailedRecentlyVisitedArticles tests', () => {
    const mockStore = configureStore();

    let wrapper, store;

    it('renders correctly', () => {
        store = mockStore(fakeData);
        const component = renderer.create(connectedComponentWithRouter(store)).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should not render spinner when data is not being resolved', () => {
        store = mockStore(fakeData);
        wrapper = render(connectedComponentWithRouter(store));

        expect(wrapper.find('.Spinner')).toHaveLength(0);
    });

    it('should render spinner when data is resolving', () => {
        const fakeStoreState = {article: {...fakeData.article, isDataBeingResolved: true}};
        store = mockStore(fakeStoreState);
        wrapper = render(connectedComponentWithRouter(store));

        expect(wrapper.find('.Spinner')).toHaveLength(1);
    });

    it('should not render article and stats when isArticleShown flag is false', () => {
        store = mockStore(fakeData);
        wrapper = render(connectedComponentWithRouter(store));

        expect(wrapper.find('.ArticleData')).toHaveLength(0);
    });

    it('should render article and stats when isArticleShown flag is true', () => {
        const fakeStoreState = {article: {...fakeData.article, isArticleShown: true}};
        store = mockStore(fakeStoreState);
        wrapper = render(connectedComponentWithRouter(store));

        expect(wrapper.find('.ArticleData')).toHaveLength(1);
    });

    it('should pass clearArticle function to Navbar component as prop', () => {
        const fakeProp = 'fakeProp';
        const fake = {article: {...fakeData.article, isArticleShown: true, clearArticle: fakeProp}};
        store = mockStore(fake);
        wrapper = shallow(pureComponentWithRouter(store, fake.article));

        const prop = wrapper.find(PureRelevantArticleData).dive().find('.Navbar li').at(0).prop('onClick');
        expect(prop).toEqual(fakeProp);
    });

});

const fakeData = {
    article: {
        isArticleShown: false,
        ...fakeRecentlyVisited,
        ...fakeArticleData,
    }
};
const connectedComponentWithRouter = (store, props) => (
    <MemoryRouter>
        <Provider store={store}>
            <RelevantArticleData {...props}/>
        </Provider>
    </MemoryRouter>
);
const pureComponentWithRouter = (store, props) => (
    <MemoryRouter>
        <PureRelevantArticleData {...props}/>
    </MemoryRouter>
);