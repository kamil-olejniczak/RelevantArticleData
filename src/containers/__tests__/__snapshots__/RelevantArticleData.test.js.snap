// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedRecentlyVisitedArticles tests renders correctly 1`] = `
<div>
  <ul
    className="Navbar"
  >
    <li
      onClick={[Function]}
    >
      Search for an article
    </li>
    <a
      href="/recentlyVisited"
      onClick={[Function]}
    >
      Your articles
    </a>
    <li>
      <a
        href="http://github.com/kamil-olejniczak"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </li>
  </ul>
  <div>
    <div
      className="ArticleSearch"
    >
      <div>
        <h1>
          Relevant Article Data
        </h1>
        <h2>
          Read only valuable content
        </h2>
      </div>
      <input
        className={null}
        onKeyDown={[Function]}
        placeholder="Paste link to article..."
        type="text"
      />
    </div>
  </div>
</div>
`;
